src/Feature/Syntax/Feature_Syntax.rei

```diff
@@ -32,6 +32,11 @@ let setTokensForLine:
 
 let update: (t, msg) => t;
 
+// [ignore(~bufferId, syntax)] marks a buffer to be ignored.
+// The only syntax highlight adjustment will come from explicit
+// calls to [setTokensForLine];
+let ignore: (~bufferId: int, t) => t;
```

Comment:
This is a very imperative API. Can the flag not be passed in with the subscription instead?

Reply:
Agreed, it would be nice to not have this imperative API at all.

 Unfortunately the syntax highlight integration (even though it has a subscription) is still imperative - the subscription manages the lifecycle of the syntax highlight service, as opposed to having per-buffer granularity (which may help streamline this). Another place we need a deeper refactor...